src/stan/lang/ast/sigs/function_signatures.hpp

```diff
@@ -257,6 +257,41 @@ namespace stan {
        */
       void add_quaternary(const::std::string& name);
 
+      /**
+       * Determine the return type of distributions' RNG function
+       * based on the primitive-ness of the arguments; if argument
+       * has dimension greater than 1 should return a vector_type.
+       *
+       * @param t type of first argument
+       */
```

Comment:
There should be a @return doc, too.  It winds up looking a bit redundant with the first line for simple functions.  

"primitiveness" doesn't have a hyphen. 

It's size, not dimension that has to be greater than 0.  But I'd just say "if both arguments are scalar, the return type is int or real depending on the distribtuion.  Otherwise, the return type is `int[]` for discrete distributions and `real[]` for continuous ones.  These aren't vector types in the Stan sense, but 1D arrays.

 If the argument has dimension greater than 0, it should return a vector type.  But I'd say that differently---if the argument is non-scalar, it will return `vector`.